src/components/MapView.jsx

```diff
@@ -131,12 +128,9 @@ export const MapView = () => {
       const isOsm = stopType === 0;
       const entryPoint = {coordinates, id, isOsm, name, ref};
 
-      if (connectionData.length === 1) {
-        if (!(connectionData[0].isOsm ^ isOsm)) {
-          dispatch(NotificationActions.error('It is not allowed to connect stops of the same type'));
-          return;
-        }
-        updateConnectionMessage(null);
+      if (connectionData.length === 1 && !(connectionData[0].isOsm ^ isOsm)) {
```

Comment:
Tego warunku nie rozumiem. `!(connectionData[0].isOsm ^ isOsm)`
O co dokładnie chodzi?

Reply:
W liście jest już jeden przystanek (`connectionData[0]`]), ma property `isOsm`. Natomiast `isOsm` na końcu linii dotyczy potencjalnego kandydata na drugi przystanek. 
Łączymy 1 przystanek z OSM i 1 spoza OSM (mogą być to różni operatorzy, np. ZTM etc.), dlatego ważne jest żeby property od jednego przystanków miało wartość inną niż ten drugi.